### PR TITLE
Fix sharding problem

### PR DESCRIPTION
**Pull request by talIguaz**
ignore 404 on `getItems` + remove commented code

### CODE REVIEW

File: pkg/utils/asynciter.go [hunk 1]
```diff
@@ -143,6 +142,13 @@ func (ic *AsyncItemsCursor) NextItem() (v3io.Item, error) {
 
 	// Read response from channel
 	resp := <-ic.responseChan
+
+	// Ignore 404s
+	if resp.Error != nil && strings.Contains(resp.Error.Error(), "404") {
```
Comment (gtopper) [inline pkg/utils/asynciter.go]: This is pretty awful... Can't we get the actual http code from `resp`?

File: pkg/utils/asynciter.go [hunk 2]
```diff
@@ -143,6 +142,13 @@ func (ic *AsyncItemsCursor) NextItem() (v3io.Item, error) {
...
+		fmt.Println("Got 404", resp.Error, "input:", resp.Request().Input)
```
Comment (gtopper) [inline pkg/utils/asynciter.go]: We really have to fix that logger...

File: pkg/utils/asynciter.go (hunk 1 above)
Comment (imakhlin) [reply pkg/utils/asynciter.go]: Need to fill feature request to HTTP-GO lib to return error codes for each HTTP call
Please check with @pavius if it's something he can do quickly

File: pkg/utils/asynciter.go (hunk 2 above)
Comment (imakhlin) [reply pkg/utils/asynciter.go]: Why not to use `ic.logger.Error(...` ?

Comment (talIguaz) [reply pkg/utils/asynciter.go]: the logger is not initialized there. so you need to either pass a logger every time you create an async iterator the problem is that one of the places doesn't have a logger to pass.
Either way our logging stuff isn't that good and Eran said he want to think what is the best practice and then notify us.. so we might as well wait with that a bit

File: pkg/utils/asynciter.go (hunk 1 above)
Comment (talIguaz) [reply pkg/utils/asynciter.go]: He is too busy, we can open a PR with a fix by ourselves but not in this PR and i guess it's not *that* urgent

Comment (gtopper) [reply pkg/utils/asynciter.go]: Ok for now, but please add a TODO and see if you can restrict this condition further (so it doesn't match something like `1534324049`).

**Pull request merged**